runtime/service.go

```diff
@@ -769,6 +785,140 @@ func (s *service) GetVMMetadata(requestCtx context.Context, request *proto.GetVM
 	return &proto.GetVMMetadataResponse{Metadata: string(metadata)}, nil
 }
 
+func (s *service) createBalloon(requestCtx context.Context, request *proto.CreateVMRequest) (models.Balloon, error) {
+	amountMB := request.BalloonDevice.AmountMb
+	deflateOnOom := request.BalloonDevice.DeflateOnOom
+	statsPollingIntervals := request.BalloonDevice.StatsPollingIntervals
+
+	balloonDevice := firecracker.NewBalloonDevice(amountMB, deflateOnOom, firecracker.WithStatsPollingIntervals(statsPollingIntervals))
+	balloon := balloonDevice.Build()
+
+	s.logger.Infof("Creating a balloon device: AmountMb=%d, DeflateOnOom=%t and statsPollingIntervals=%d ", *balloon.AmountMb, *balloon.DeflateOnOom, balloon.StatsPollingIntervals)
+	return balloon, nil
+}
+
+func (s *service) BuildBalloonDeviceOpt(balloon models.Balloon) ([]firecracker.Opt, error) {
```

Comment:
Does it have to be exported to other packages?

Reply:
Nope. Fixed it.